File: examples/scripts/hello.lua
-- hello.lua — Minimal Bolt Scan example script (Lua)
--
-- Usage contract: Bolt Scan passes <target> and <port> as command-line
-- arguments. Print a single line to stdout as the script result.
--
-- Run manually: lua hello.lua 192.168.1.1 80

local target = arg[1]
local port   = arg[2]
print("Hello from Lua: " .. target .. ":" .. port)

File: examples/scripts/hello.rs
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 3 {
        println!("Usage: {} <target> <port>", args[0]);
        std::process::exit(1);
    }
    println!("Hello from Rust: {}:{}", args[1], args[2]);
}
